src/fr-auth/index.ts

```diff
@@ -67,9 +71,102 @@ abstract class FRAuth {
     return new FRLoginFailure(nextPayload);
   }
+
+  /**
+   * Redirects to the URL identified in the RedirectCallback and saves the full
+   * step information to localStorage for retrieval when user returns from login.
+   *
+   * Example:
+   * ```js
+   * forgerock.FRAuth.redirect(step);
+   * ```
+   */
+  public static redirect(step: FRStep): void {
+    const cb = step.getCallbackOfType(CallbackType.RedirectCallback) as RedirectCallback;
+    const redirectUrl = cb.getRedirectUrl();
+
+    window.localStorage.setItem(this.previousStepKey, JSON.stringify(step));
+    window.location.assign(redirectUrl);
+  }
+
+  /**
+   * Resumes a tree after returning from an external client or provider.
+   * Requires the full URL of the current window. It will parse URL for
+   * key-value pairs as well as, if required, retrieves previous step.
+   *
+   * Example;
+   * ```js
+   * forgerock.FRAuth.resume(window.location.href)
+   * ```
+   */
+  public static async resume(
+    url: string,
+    options?: StepOptions,
+  ): Promise<FRStep | FRLoginSuccess | FRLoginFailure> {
+    const parsedUrl = new URL(url);
+    const code = parsedUrl.searchParams.get('code');
+    const nonce = parsedUrl.searchParams.get('nonce');
+    const state = parsedUrl.searchParams.get('state');
+    const suspendedId = parsedUrl.searchParams.get('suspendedId');
+
+    let previousStep;
+
+    if (code && state) {
+      /**
+       * If we are returning back from a provider, and code and state are present,
+       * this is a continuation of Social Login, so retrieve the previous step from
+       * localStorage. Then delete it to remove stale data.
+       */
+      const redirectStepString = window.localStorage.getItem(this.previousStepKey);
+
+      if (!redirectStepString) {
+        throw new Error('Error: could not retrieve original redirect information.');
+      }
+
+      try {
+        previousStep = JSON.parse(redirectStepString);
+      } catch (err) {
+        throw new Error('Error: could not parse redirect params or step information');
+      }
+
+      window.localStorage.removeItem(this.previousStepKey);
+    }
+
+    /**
+     * Construct options object from the options parameter and key-value pairs from URL.
+     * Ensure query parameters from current URL are the last properties spread in the object.
+     */
+    const nextOptions = {
+      ...options,
+      query: {
+        ...(options && options.query),
+        ...(code && { code }),
+        ...(nonce && { nonce }),
+        ...(state && { state }),
+        ...(suspendedId && { suspendedId }),
+      },
+    };
+
+    const step = await this.next(previousStep, nextOptions);
+    return step;
+  }
+
+  /**
+   * Requests the first step in the authentication tree.
+   * This is essentially an alias to calling FRAuth.next without a previous step.
+   *
+   * @param options Configuration overrides
+   * @return The next step in the authentication tree
+   */
+  public static async start(
```

Comment:
This newly added method for SDK alignment is just essentially an alias to `FRAuth.next` in order to align with native mobile.